lib/dockerregistry/inventory.go

```diff
@@ -1636,14 +1653,14 @@ func (sc *SyncContext) ExecRequests(
 	go func() {
 		for reqRes := range requestResults {
 			if len(reqRes.Errors) > 0 {
+				klog.Errorf(
+					"Request %v: error(s) encountered: %v\n",
+					reqRes.Context, reqRes.Errors)
 				(*mutex).Lock()
 				err = fmt.Errorf("Encountered an error during the" +
 					" promotion step")
+				sc.CollectErrors(reqRes.Errors...)
```

Comment:
I think you can replace this line with a call to `append()` directly. I'm pretty sure it can accept a variable number of arguments.